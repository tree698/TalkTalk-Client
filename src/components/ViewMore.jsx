import React from 'react';
import { BsCheckLg } from 'react-icons/bs';
import Banner from './ui/Banner';

export default function ViewMore({ lengthDrawings, limit, onButtonClick }) {
  return (
    <div>
      {lengthDrawings === limit ? (
        <button
          onClick={() => onButtonClick()}
          className="w-[200px] md:w-[400px] py-1 md:py-2 lg:py-3 px-2 mt-8 md:mt-10 lg:mt-12 mb-8 text-xl font-semibold text-center text-white bg-accent rounded-2xl hover:scale-105 hover:shadow-xl transition-all delay-150 duration-300 ease-in-out"
        >
          View More
        </button>
      ) : (
        <div className="flex items-center gap-4 mt-12 text-2xl">
          <BsCheckLg />
          <Banner text="No more drawings" />
        </div>
      )}
    </div>
  );
}
