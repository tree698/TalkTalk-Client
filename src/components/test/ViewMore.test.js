import ViewMore from '../ViewMore';
import renderer from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ViewMore', () => {
  it('renders with button', () => {
    const component = renderer.create(
      <ViewMore lengthDrawings={3} limit={3} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders without button', () => {
    const component = renderer.create(
      <ViewMore lengthDrawings={2} limit={3} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('send client event', async () => {
    const mockOnClick = jest.fn();
    render(
      <ViewMore lengthDrawings={3} limit={3} onButtonClick={mockOnClick} />
    );

    const button = screen.getByRole('button');
    userEvent.click(button);

    await waitFor(() => {
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
