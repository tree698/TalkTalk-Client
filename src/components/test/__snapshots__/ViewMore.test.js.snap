// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMore renders with button 1`] = `
<div>
  <button
    className="w-52 md:w-56 lg:w-60 py-1 md:py-2 lg:py-3 my-6 md:my-8 lg:my-10  text-xs md:text-sm lg:text-base font-semibold text-center text-white bg-brand rounded-3xl hover:scale-105 hover:shadow-xl transition-all delay-150 duration-300 ease-in-out"
    onClick={[Function]}
  >
    View More
  </button>
</div>
`;

exports[`ViewMore renders without button 1`] = `
<div>
  <div
    className="flex items-center gap-1 md:gap-2 lg:gap-3 my-6 md:my-8 lg:my-10 text-xs md:text-sm lg:text-base"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.736 3.97a.733.733 0 0 1 1.047 0c.286.289.29.756.01 1.05L7.88 12.01a.733.733 0 0 1-1.065.02L3.217 8.384a.757.757 0 0 1 0-1.06.733.733 0 0 1 1.047 0l3.052 3.093 5.4-6.425a.247.247 0 0 1 .02-.022Z"
      />
    </svg>
    <p>
      No more drawings
    </p>
  </div>
</div>
`;
